presto-main/src/main/java/com/facebook/presto/server/AsyncPageTransportForwardFilter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ *  It is not possible to map the {@link AsyncPageTransportServlet} to "/v1/task/<taskid>/results/*" directly.
+ *  Servlet pattern matching doesn't allow arbitrary globbing (e.g.: /v1/task/*\/results/*).
+ *  Only prefix or suffix matching is allowed (e.g.: /v1/task/*, *\/suffix).
+ *  Hence a more nuanced matching and a forward is required.
+ */
+public class AsyncPageTransportForwardFilter
```

Comment:
Why not just implement the rename in one shot, as opposed to forwarding `/v1/task` to `/v1/task/async`?  i.e. why not just use `/v1/task` and remove this filter?

Reply:
Then all the requests starting with `/v1/task` will be routed to the custom servlet, while requests other than `/v1/task*/results/*` have to be routed to `TaskResource`

Reply:
I can't think of a better way of doing it.